include/gsl/util

```diff
@@ -52,40 +52,29 @@ template <class F>
 class final_action
 {
 public:
-    static_assert(!std::is_reference<F>::value && !std::is_const<F>::value &&
-                      !std::is_volatile<F>::value,
-                  "Final_action should store its callable by value");
+    template <class FF>
+    explicit final_action(FF&& ff) noexcept : f{std::forward<FF>(ff)} { }
 
-    explicit final_action(F f) noexcept : f_(std::move(f)) {}
+    ~final_action() noexcept { if (invoke) f(); }
 
     final_action(final_action&& other) noexcept
-        : f_(std::move(other.f_)), invoke_(std::exchange(other.invoke_, false))
-    {}
+        : f(std::move(other.f)), invoke(std::exchange(other.invoke, false))
+    { }
 
-    final_action(const final_action&) = delete;
-    final_action& operator=(const final_action&) = delete;
-    final_action& operator=(final_action&&) = delete;
-
-    // clang-format off
-    GSL_SUPPRESS(f.6) // NO-FORMAT: attribute // terminate if throws
-    // clang-format on
-    ~final_action() noexcept
-    {
-        if (invoke_) f_();
-    }
+    final_action(const final_action& rhs) = delete;
+    void operator=(const final_action&)   = delete;
```

Comment:
missing parameter name? (alternatively, remove the other two names)

Reply:
Fixed, thanks!